### ISSUE TITLE
Unexpected nullability warning for Nullable<int>.ToString()

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#49129]
**Version Used**: 16b67aa0

### Steps to Reproduce

```csharp
string value = default(int?).ToString();
```

### Expected Behavior

No diagnostics (`Nullable<int>.ToString()` cannot return `null`).

### Actual Behavior

CS8600 is reported.

### Additional details

Given that both https://github.com/dotnet/runtime/issues/1318 and https://github.com/dotnet/roslyn/pull/45759 were rejected, the compiler needs to create and/or document a solution before this false positive can be eliminated.

**Comment by jaredpar** [comment]
Closing as the compiler is operating as designed here. The runtime decided to make the `ToString` signature return `string?` and hence the compiler is correctly respecting that API decision. The customer has a variety of techniques there can apply here from suppression `!` to `?? ""` or you can follow up with the runtime team.
